src/MPQEditor/MPQEditor.ts

```diff
@@ -0,0 +1,525 @@
+          );
+          break;
+        case "updateDocument":
+          this.updateDocument(document);
+          break;
+        case "removeLayer":
+          this.handleRemoveLayerFromLayers(e.name, document);
+          break;
+        default:
+          break;
+      }
+    });
+  }
+
+  /**
+   * Update document by text
+   */
+  public static async updateDocumentBy(
+    document: vscode.TextDocument,
+    text: string
+  ) {
+    const edit = new vscode.WorkspaceEdit();
+    edit.replace(
+      document.uri,
+      new vscode.Range(0, 0, document.lineCount, 0),
+      text
+    );
+    await vscode.workspace.applyEdit(edit);
+  }
+
+  /**
+   * Update document
+   */
+  private async updateDocument(document: vscode.TextDocument) {
+    if (
+      this._mpqDataMap[document.uri.toString()].getAsString() !==
+      document.getText()
+    ) {
+      MPQEditorProvider.updateDocumentBy(
+        document,
+        this._mpqDataMap[document.uri.toString()].getAsString()
+      );
+    }
+  }
+
+  /**
+   * Initializes webviewpanel
+   */
+  private initWebviewPanel(
+    document: vscode.TextDocument,
+    webviewPanel: vscode.WebviewPanel
+  ): void {
+    vscode.commands.executeCommand(
+      "setContext",
+      MPQEditorProvider.viewType,
+      true
+    );
+
+    const changeDocumentSubscription = vscode.workspace.onDidChangeTextDocument(
+      (e) => {
+        if (
+          e.contentChanges.length > 0 &&
+          e.document.uri.toString() === document.uri.toString()
+        ) {
+          this.updateWebview(document, webviewPanel.webview);
+        }
+      }
+    );
+
+    webviewPanel.onDidChangeViewState(
+      () => {
+        vscode.commands.executeCommand(
+          "setContext",
+          MPQEditorProvider.viewType,
+          webviewPanel.visible
+        );
+      },
+      null,
+      this._disposables
+    );
+
+    webviewPanel.onDidDispose(() => {
+      changeDocumentSubscription.dispose();
+      while (this._disposables.length) {
+        const x = this._disposables.pop();
+        if (x) {
+          x.dispose();
+        }
+      }
+      vscode.commands.executeCommand(
+        "setContext",
+        MPQEditorProvider.viewType,
+        false
+      );
+    });
+  }
+
+  /**
+   * Update webview with document
+   */
+  private updateWebview(
+    document: vscode.TextDocument,
+    webview: vscode.Webview
+  ): void {
+    this._mpqDataMap[document.uri.toString()].setWithString(document.getText());
+    const content = JSON.parse(document.getText());
+    if (content !== undefined) {
+      webview.postMessage({
+        type: "displayMPQ",
+        content: content,
+      });
+    }
+  }
+
+  /**
+   * Add layers to edit their quantization parameters
+   */
+  private hadleAddSpecificLayerFromDialog(document: vscode.TextDocument) {
+    const nodes =
+      this._mpqDataMap[document.uri.toString()].getDefaultModelLayers();
+    const pickOptions = {
+      canPickMany: true,
+    };
```

Comment:
How about applying those quickPickOptions? 😄 

```suggestion
    const pickOptions = {
      title: "Add specific layers to edit their quantization parameters",
      matchOnDescription: true,
      matchOnDetail: true,
      placeHolder: "Select layers",
      ignoreFocusOut: true,
      canPickMany: true,
    };
```

Reply:
![190e428b-2d12-4309-a964-d204e1629614](https://user-images.githubusercontent.com/17171963/229042890-17182cd9-c14d-49e5-bb77-e943508dbb31.gif)

Reply:
it will
- title: show a title "Add specific~"
- placeHolder: show a place holder "Select layers"
- ignoreFocusOut: stay even if you clicks other sides
- matchOnDescription/Detail: (well it won't affect cuz we don't have description/details in the list elements)

Reply:
Ahhh. Thank you very much!